internal/storage/gcs/bucket.go

```diff
@@ -20,6 +20,36 @@ import (
 	"golang.org/x/net/context"
 )
 
+// ObjectWriter interface represents asynchronous object
+// writers, which don't write the entire GCS object from
+// content in a single-shot. These objects are created once,
+// are written to multiple times and are eventually closed
+// once. On closing, they return a GCS object (gcs.Object)
+// and and error object for err-handling.
+// TODO: rename this to AsynchronousGcsObjectWriter
+type ObjectWriter interface {
+	// Write writes the entire content by reading from the given
+	// reader interface.
+	// io.EOF is an expected error in case the reader has less
+	// data than the writer tries to read from it. When EOF is
+	// returned and whether it's an actionable error, depends on
+	// the actual implementation and invocation of this interface.
+	// In other words, EOF might not be a fatal error in some cases.
+	// Unlike the io.Writer interface, it doesn't return number-of-bytes
+	// written so far, as the write is asynchronous; for that we have the
```

Comment:
wrong comment. write will never the number of bytes written so far, it only returns the number of bytes written in that particular call.

Reply:
Corrected. PTAL.